tidy3d/components/geometry.py

```diff
@@ -1142,74 +1366,393 @@ def _find_intersecting_vertices(
             Integer index into 'xyz' (0,1,2).
 
         Returns
-        np.ndarray, np.ndarray
-            Backward (xy) vertices and forward (xy) vertices.
+        np.ndarray
+            Height (relative to the base) where the plane will intersect with vertices.
         """
+        if np.isclose(self.sidewall_angle, np.pi / 2):
+            return np.array([self.length])
+
+        vertices = self.base_polygon.copy()
+
+        # shift rate
+        dist = 1.0
+        shift_x, shift_y = PolySlab._shift_vertices(vertices, dist)[2]
+        shift_val = shift_x if axis == 0 else shift_y
+        shift_val[np.isclose(shift_val, 0)] = np.inf  # for static vertices
+
+        # distance to the plane in the direction of vertex shifting
+        distance = vertices[:, axis] - position
+        height = distance * np.tan(self.sidewall_angle) / shift_val
+        height = np.unique(height)
+        if self.sidewall_angle > np.pi / 2:
+            height *= -1
+
+        height = height[height > 0]
+        height = height[height < self.length]
+        return height
+
+    def _find_intersecting_ys_angle_vertical(  # pylint:disable=too-many-locals
+        self, vertices: np.ndarray, position: float, axis: int
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """Finds pairs of forward and backwards vertices where polygon intersects position at axis,
+        Find intersection point (in y) assuming straight line,and intersecting angle between plane
+        and edges. (For unslanted polyslab)
+           Assumes axis is handles so this function works on xy plane.
 
-        vertices_b = np.array(self.vertices)
+        Parameters
+        ----------
+        vertices : np.ndarray
+            Shape (N, 2) defining the polygon vertices in the xy-plane
```

Comment:
periods at the end of these descriptions

Reply:
fixed